core/src/test/java/org/apache/iceberg/TestMetadataTableScans.java

```diff
@@ -516,6 +522,101 @@ public void testPartitionColumnNamedPartition() throws Exception {
     validateIncludesPartitionScan(tasksAndEq, 0);
   }
 
+  @Test
+  public void testPartitionsTableScanWithDeleteFilesInFilter() throws IOException {
+    Assume.assumeTrue(formatVersion == 2);
+    Configuration conf = new Configuration();
+    HadoopTables tables = new HadoopTables(conf);
```

Comment:
Seems all the rest of the test in this class just use the constructor of  PartitionsTable (or whatever metadata table), is there any reason not to do it here?

Reply:
I have tried to use `new PartitionTable(table.ops(), table)`, but it return a NPE on `ops.current().metadataFileLocation()`. https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/PartitionsTable.java#L84

Reply:
That's strange, actually it works in most of the tests.  Do you do it after populating the table, like the other tests?

Reply:
Yes, I checked again. I think the problem is that TestTableOperations does not write metadata.json file:https://github.com/apache/iceberg/blob/master/core/src/test/java/org/apache/iceberg/TestTables.java#L216.
So the metadata location is always null. When reading the records. it should plan tasks,so the metadata.json should be used [here](https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/PartitionsTable.java#L78).
So I create a HadoopTable because HadoopTableOperation will create the metadata.json [here](https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/hadoop/HadoopTableOperations.java#L151)